### PR TITLE
Make preconditioners the identity when skipped

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Add an action that makes an iterative linear solve the identity operation
if no steps were performed at all, i.e. assume A=1 so Ax=b solves to x=b.
This is useful when the solver is used as preconditioner, because then we
can disable preconditioning by just not iterating the preconditioner, i.e.
by setting its number of iterations to 0 in the input file. The alternative 
(when not using this action) is to keep the fields at their initial guess x=x0.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `major new feature` if appropriate.

### CODE REVIEW

File: src/ParallelAlgorithms/LinearSolver/AsynchronousSolvers/ElementActions.hpp [hunk 1]
```diff
@@ -196,6 +196,21 @@ struct PrepareSolve {
     contribute_to_residual_observation<OptionsGroup, ParallelComponent>(
         iteration_id, residual_magnitude_square, cache, array_index);
 
+    // Make the solve the identity operation if no steps are going to be
```
Comment (kidder) [inline src/ParallelAlgorithms/LinearSolver/AsynchronousSolvers/ElementActions.hpp]: Please make this handled by a new flag and not the number of iterations.  I would absolutely not expect this behavior

**Review by kidder — CHANGES_REQUESTED** [review]

**Comment by nilsvu** [comment]
Thanks for pointing that out @kidder, I converted this feature to an action.

**Comment by kidder** [comment]
is the db::mutate in the new action tested by your changes to the test code, or do the tests so far skip that code?

**Comment by nilsvu** [comment]
@kidder no the tests don't currently run hit the condition. I can add a separate action unit test though.

**Comment by nilsvu** [comment]
@kidder added the test

**Review by kidder — APPROVED** [review]

**Review by fmahebert — APPROVED** [review]

**Pull request merged**